File: Cpp/Source/TextConverter.Test/source/HtmlTextConverterTest.cpp

#include "TextConverter/include/HtmlTextConverter.h"

#include <cppunit/TestCase.h>
#include <cppunit/TestFixture.h>
#include <cppunit/extensions/HelperMacros.h>

class HtmlTextConverterTest : public CppUnit::TestFixture 
{
public:
    virtual void setUp() {}
    virtual void tearDown() {}

    void t_foo();
    
    CPPUNIT_TEST_SUITE(HtmlTextConverterTest);
    CPPUNIT_TEST(t_foo);
    CPPUNIT_TEST_SUITE_END();
};

void HtmlTextConverterTest::t_foo()
{
    HtmlTextConverter converter("foo");
    CPPUNIT_ASSERT( std::string("fixme").compare( converter.getFilename() ) == 0 );
}

CPPUNIT_TEST_SUITE_REGISTRATION(HtmlTextConverterTest);

File: php/tests/TextConverter/HtmlTextConverterTest.php
<?php

declare(strict_types=1);

namespace Tests\TextConverter;

use PHPUnit\Framework\TestCase;
use RacingCar\TextConverter\HtmlTextConverter;

class HtmlTextConverterTest extends TestCase
{
    public function testFoo(): void
    {
        $converter = new HtmlTextConverter('foo');
        $this->assertSame('fixme', $converter->getFileName());
    }
}
